### PR TITLE
Port utils::convert_text() and UTF-8 <-> locale converters to Rust

### PR DESCRIPTION
**Pull request by Minoru**
This is another stepping stone to #1344.

I had to add my own bindings for `iconv()` because [my PR to the libc crate](https://github.com/rust-lang/libc/pull/2037) is stuck due to linking issues of macOS. In Newsboat, I have more control over the build process than I have in the libc crate, so I managed to overcome those issues here; but we should switch to libc bindings once that PR is merged.

There are also some trouble with tests. https://github.com/newsboat/newsboat/pull/1443 added tests that require some locales to be present. Users and distro's build machines probably don't have those, so I put in checks to skip those tests. I couldn't transfer this trick to Rust directly, so Rust uses a slightly different scheme. More on that in the commit message to https://github.com/newsboat/newsboat/commit/7ec21f983f905efb182afafaf09f724bfb6fe0ba

A commit message is not a good place for docs, so I also brushed up the hacker's guide a bit: marked most sections as outdated, but updated the one on tests. I don't yet have a plan on how to fully restore the hacker's guide, and what I actually want out of that document, but it's a start.

Reviews are welcome! Don't be afraid of large number of changes — it's the usual "rewrite in C++, port tests" stuff, nothing hardcore. I'll merge this in three days if no issues are raised.

### CODE REVIEW

File: doc/internal/hackers-guide.asciidoc [hunk 1]
```diff
@@ -122,6 +131,9 @@ crude menus that are somehow hard to use.
 
 === Keep a good balance of customizability
 
+**Status of this section:** the idea holds, but Newsboat arguably failed this
+goal (January 2021).
+
 The problem with user wishes is that too many people demand a possibility to
 customize this bell or that whistle within newsbeuter. Often, these
```
Comment (der-lyse) [inline doc/internal/hackers-guide.asciidoc:138]: Not sure whether this is on purpose, to signal the age, but it's still called "newsbeuter". The same goes for a few other lines I'm not able to comment on. :-)

File: doc/internal/hackers-guide.asciidoc [hunk 2]
```diff
@@ -175,17 +195,31 @@ code, so it's only helpful for developers.
...
+
+C++ tests are in the _test_ subdirectory. They're using
+https://github.com/catchorg/Catch2/tree/devel/docs[the Catch2 framework]. Rust
+tests are split: unit tests are in `mod tests` sub-modules under
+_rust/libnewsboat/src_, while integration tests are under
+_rust/libnewsboat/tests_. Rust's test runner is multi-threaded, so we use
+integration testing when we want to do the checks in a separate process.
+
+In addition to `TMPDIR=/dev/shm make -j5 PROFILE=1 check` incantation that's
+already explained in the contributing guidelines, you should know about
```
Comment (der-lyse) [inline doc/internal/hackers-guide.asciidoc:208]: …about **the**…

**Review by der-lyse — COMMENTED** [review]
Nice!

File: doc/internal/hackers-guide.asciidoc (hunk 1 above)
Comment (Minoru) [reply doc/internal/hackers-guide.asciidoc:138]: It's on purpose: I didn't want to get distracted updating the sections that aren't relevant to this particular PR. I'm also not yet sure how I want the document to look, so I don't have an immediate plan to rephrase all the old sections and update them.

Perhaps I should explicitly say "outdated", to make it even more clear that the sections wasn't updated?

File: doc/internal/hackers-guide.asciidoc (hunk 2 above)
Comment (Minoru) [reply doc/internal/hackers-guide.asciidoc]: Thanks! Fixed in a33be00d8b7a60ab13cf37caf2bf41d08fe1b2a2.

File: doc/internal/hackers-guide.asciidoc (hunk 1 above)
Comment (der-lyse) [reply doc/internal/hackers-guide.asciidoc:138]: I'm absolutely fine with leaving it like that, no need to change anything. Just wanted to double-check.

**Pull request merged**